freight/auth/__init__.py

```diff
@@ -0,0 +1,37 @@
+from .exceptions import AccessDeniedError, ProviderConfigurationError  # noqa
+from .providers import GoogleOAuth2Provider, GitHubOAuth2Provider
+
+
+class Auth(object):
+    """Flask extension for OAuth 2.0 authentication.
+
+    Assigns the configured provider to the ``auth_provider`` key of
+    ``app.state`` when initialized.
+    """
+
+    def __init__(self, app=None):
+        if app is not None:
+            self.init_app(app)
+
+    def init_app(self, app):
+        # For compatibility with previous versions of Freight, default to using
+        # Google as the authentication backend.
+        backend = app.config.get('AUTH_BACKEND')
+        if not backend:
+            backend = 'google'
+
+        # Resolve the provider setup function.
+        try:
+            provider_cls = {
```

Comment:
this is good, and if we want to extend it in the future we can just have AUTH_BACKEND support a full class path or similar